lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1668,87 +1628,87 @@ class SemanticsObject {
     _currentChildrenInRenderOrder = childrenInRenderOrder;
   }
 
-  /// The primary role of this node.
+  /// The role of this node.
   ///
-  /// The primary role is assigned by [updateSelf] based on the combination of
+  /// The role is assigned by [updateSelf] based on the combination of
   /// semantics flags and actions.
-  PrimaryRoleManager? primaryRole;
+  SemanticRole? semanticRole;
 
-  PrimaryRole _getPrimaryRoleIdentifier() {
+  SemanticRoleId _getSemanticRoleId() {
```

Comment:
Should this be a `SemanticRole.kind` that every SemanticRole needs to override?

Reply:
Done. Made it `SemanticRoleKind`.